expression/builtin_miscellaneous_vec.go

```diff
@@ -334,8 +334,15 @@ func (b *builtinSleepSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) e
 }
 
 func doSleep(secs float64, sessVars *variable.SessionVars) (isKilled bool) {
+	if secs <= 0.0 {
+		return false
+	}
 	dur := time.Duration(secs * float64(time.Second.Nanoseconds()))
-	ticker := time.NewTicker(10 * time.Millisecond)
+	d := 10 * time.Millisecond
+	if dur < d {
+		d = dur
+	}
+	ticker := time.NewTicker(d)
```

Comment:
Hi, the old code seems to be not quite accurate. If we want to sleep 11ms, it may always sleep 20ms.
I think what best to do is adding a `time.Timer` to control the end of sleep while keeping` time.Ticker` to control the signal of a kill.

```go
ticker := time.NewTicker(10 * time.Millisecond)
timer := time.NewTimer(dur) 
for {
    select {
    case <-ticker.C:
        if  killed {
            timer.Stop()
            return false
        }
    case <-timer.C:
        return true
    }
}
```

Reply:
After will this logic, the test `TestVectorizedBuiltinMiscellaneousEvalOneVec ` and `TestVectorizedBuiltinMiscellaneousFunc ` will cost nearly 0.8s in compare to the origin code will only one time both cost about 0.1s. I'm not sure if this will be a performance problem. @SunRunAway  PTAL

Reply:
I think it's acceptable.